### PR TITLE
CLN: tests/extension/json/test_json.py typefix

### PR DESCRIPTION
**Pull request by lukasbk**
The error is due to the fact that the builtins.staticmethod type defined in extension.base.base.BaseExtensionTests does not match the function type in extension.json.test_json.BaseJSON. I don't see a possibility for a proper type hint fix without changing the structure of the code. One way would be to replace the call of staticmethod by function definitions that call the corresponding methods.
Here I propose an easier suppression of the error using type: Any.
What do you think?

- [ ] xref #28926
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]
@lukasbk Thanks for the PR.

This PR also has the changes from #28990, can you keep these separate.

also see #28988 re discussion on solving `Definition of "assert_series_equal" in base class "BaseJSON" is incompatible with definition in base class "BaseExtensionTests"` errors.

**Comment by jbrockmendel** [comment]
@lukasbk can you rebase

File: pandas/tests/extension/base/base.py [hunk 1]
```diff
@@ -2,8 +2,18 @@
 
 
 class BaseExtensionTests:
+    @staticmethod
+    def assert_equal(left, right, **kwargs):
+        return tm.assert_equal(left, right, **kwargs)
```
Comment (simonjayhawkins) [inline pandas/tests/extension/base/base.py]: don't need the return

File: pandas/tests/extension/base/base.py [hunk 2]
```diff
@@ -2,8 +2,18 @@
...
 
-    assert_equal = staticmethod(tm.assert_equal)
-    assert_series_equal = staticmethod(tm.assert_series_equal)
-    assert_frame_equal = staticmethod(tm.assert_frame_equal)
-    assert_extension_array_equal = staticmethod(tm.assert_extension_array_equal)
+    @staticmethod
+    def assert_series_equal(left, right, **kwargs):
+        return tm.assert_series_equal(left, right, **kwargs)
```
Comment (simonjayhawkins) [inline pandas/tests/extension/base/base.py]: same

File: pandas/tests/extension/base/base.py [hunk 3]
```diff
@@ -2,8 +2,18 @@
...
+
+    @staticmethod
+    def assert_frame_equal(left, right, **kwargs):
+        return tm.assert_frame_equal(left, right, **kwargs)
```
Comment (simonjayhawkins) [inline pandas/tests/extension/base/base.py]: same

File: pandas/tests/extension/base/base.py [hunk 4]
```diff
@@ -2,8 +2,18 @@
...
+
+    @staticmethod
+    def assert_extension_array_equal(left, right, **kwargs):
+        return tm.assert_extension_array_equal(left, right, **kwargs)
```
Comment (simonjayhawkins) [inline pandas/tests/extension/base/base.py]: same

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]

File: pandas/tests/extension/base/base.py [hunk 5]
```diff
@@ -2,8 +2,18 @@
 
 
 class BaseExtensionTests:
+    @staticmethod
```
Comment (WillAyd) [inline pandas/tests/extension/base/base.py]: Can you get rid of these methods altogether and just import / use as function within tests?

**Review by WillAyd — CHANGES_REQUESTED** [review]

**Comment by WillAyd** [comment]
@lukasbk looks like a merge conflict; can you fix up and address comments?

File: pandas/tests/extension/base/base.py (hunk 5 above)
Comment (simonjayhawkins) [reply pandas/tests/extension/base/base.py]: I don't think so, since some tests need to override these

File: pandas/tests/extension/base/base.py [hunk 6]
```diff
@@ -2,8 +2,18 @@
...
+    def assert_equal(left, right, **kwargs):
```
Comment (simonjayhawkins) [inline pandas/tests/extension/base/base.py]: AFAICT it's only `assert_series_equal` and `assert_frame_equal` that currently need to be overridden in specific extension array tests, so `assert_equal` and `assert_extension_array_equal` don't need to be changed at this time.

**Comment by jreback** [comment]
looks ok to me. I am bit concerned that we are now changing back to using ``tm`` as we *do* want the ability to override on these tests, but I guess that mypy will check this now.

File: pandas/tests/extension/base/base.py (hunk 5 above)
Comment (WillAyd) [reply pandas/tests/extension/base/base.py]: Hmm OK - so was this change required? Looks like using `tm.assert_*` in the tests now?

**Comment by jreback** [comment]
can you rebase

**Comment by WillAyd** [comment]
@lukasbk can you fix up the merge conflict? I think we can get this in if so

**Comment by datapythonista** [comment]
@lukasbk the CI didn't pass, and this is conflicting again.

Do you mind merging master again, making sure the CI passes, and pinging us to get this merged quickly after that please? Thanks!

**Comment by WillAyd** [comment]
@lukasbk can you fix up merge conflict? I think this was close

**Comment by simonjayhawkins** [comment]
closing as stale. cc @SaturnFromTitan

**Pull request closed without merging**